src/commons/asScrollableContent.tsx

```diff
@@ -0,0 +1,82 @@
+import _ from 'lodash';
+import React, {Component} from 'react';
+import {ScrollView, ScrollViewProps} from 'react-native';
+//@ts-ignore
+import hoistStatics from 'hoist-non-react-statics';
+
+interface ScrollableContentProps {
+  scrollableContentProps?: ScrollViewProps;
+  maximumContentSize?: number;
+}
+
+interface ScrollableContentState {
+  maximumContentSize?: number;
+  contentSize?: number;
+  scrollEnabled: boolean;
+}
+
+type PropTypes = ScrollableContentProps;
+
+export default function asScrollableContent(WrappedComponent: React.ComponentType<any>) {
+  class ScrollableContent extends Component<PropTypes, ScrollableContentState> {
+    static displayName: string | undefined;
+
+    constructor(props: PropTypes) {
+      super(props);
+      this.state = {
+        maximumContentSize: props.maximumContentSize,
+        contentSize: undefined,
+        scrollEnabled: false
```

Comment:
we have the scrollEnable state to lock/unlock the scroll that's only internal. 
How would you use this HOC to decide if the to render a gradient at the bottom? or anything at all. 
You don't pass the user back this value in order to make a use of it.